amortization.py

```diff
@@ -41,19 +48,35 @@ def pmt(rate, nper, pv, typ=0):
 def presentValueOfAnnuity(cflw, rate, nper):
     return cflw * ((1 - (1 + rate)**(-nper)) / rate)
 
+def _cast_(castme, typed):
+    if isinstance(typed, Decimal):
+        return Decimal(castme)
+    else:
+        return float(castme)
+
 def calculate_interest(rate, base, start_date=None, end_date=None):
     """ Calculate interest for stated (yearly) interest rate "rate"
         according to static variables defining method """
+    assert(start_date < end_date)
     if (COMPOUNDING_PERIOD == MONTHLY_COMPOUNDING and
             BILLING_PERIOD == MONTHLY_BILLING):
         return _typeless_round(rate * base / 12)
     if (COMPOUNDING_PERIOD == SEMIMONTHLY_COMPOUNDING and
             BILLING_PERIOD == SEMIMONTHLY_BILLING):
         return _typeless_round(rate * base / 24)
-    if COMPOUNDING_PERIOD == DAILY_COMPOUNDING:
+    if COMPOUNDING_PERIOD == ACTUAL_360_COMPOUNDING:
         days_elapsed = (end_date - start_date).days
         interest = (((rate/360)+1)**(days_elapsed) * base) - base
         return _typeless_round(interest)
+    if COMPOUNDING_PERIOD == ACTUAL_ACTUAL_COMPOUNDING:
```

Comment:
since everything is simple interest, thinking we keep this designator out of the constant (to avoid confusion); I can put a comment at the top of the file to clarify.